Connectors/Autocad/Speckle.Connectors.Civil3dShared/HostApp/PropertySetBaker.cs

```diff
@@ -0,0 +1,369 @@
+    Dictionary<string, object?> setData,
+    ADB.Transaction tr
+  )
+  {
+    try
+    {
+      if (!entity.IsWriteEnabled)
+      {
+        entity.UpgradeOpen();
+      }
+
+      AAECPDB.PropertyDataServices.AddPropertySet(entity, propertySetDefId);
+
+      return UpdatePropertySet(entity, propertySetDefId, setData, tr);
+    }
+    catch (Exception ex) when (!ex.IsFatal())
+    {
+      _logger.LogWarning(ex, "Failed to add property set to entity");
+      return false;
+    }
+  }
+
+  private bool UpdatePropertySet(
+    ADB.Entity entity,
+    ADB.ObjectId propertySetDefId,
+    Dictionary<string, object?> setData,
+    ADB.Transaction tr
+  )
+  {
+    try
+    {
+      ADB.ObjectId propertySetId = AAECPDB.PropertyDataServices.GetPropertySet(entity, propertySetDefId);
+      var propertySet = (AAECPDB.PropertySet)tr.GetObject(propertySetId, ADB.OpenMode.ForWrite);
+      var setDefinition = (AAECPDB.PropertySetDefinition)tr.GetObject(propertySetDefId, ADB.OpenMode.ForRead);
+
+      // Build a map of property names to definition IDs
+      Dictionary<string, int> propertyNameToId = new();
+      foreach (AAECPDB.PropertyDefinition propDef in setDefinition.Definitions)
+      {
+        propertyNameToId[propDef.Name] = propDef.Id;
+      }
+
+      foreach (var propertyEntry in setData)
+      {
+        string propertyName = propertyEntry.Key;
+        object? propertyDataObj = propertyEntry.Value;
+
+        if (propertyDataObj is not Dictionary<string, object?> propertyDataDict)
+        {
+          continue;
+        }
+
+        if (!propertyDataDict.TryGetValue("value", out var value) || value == null)
+        {
+          continue;
+        }
+
+        if (!propertyNameToId.TryGetValue(propertyName, out int propertyId))
+        {
+          continue;
+        }
+
+        _propertyHandler.TryGetValue(
+          () =>
+          {
+            propertySet.SetAt(propertyId, value);
+            return true;
+          },
+          out _
+        );
+      }
+
+      return true;
+    }
+    catch (Exception ex) when (!ex.IsFatal())
+    {
+      _logger.LogWarning(ex, "Failed to update property set values");
+      return false;
+    }
+  }
+
+  private bool TryParseDataType(string dataTypeString, out AAEC.PropertyData.DataType dataType)
+  {
+    return Enum.TryParse(dataTypeString, out dataType);
+  }
+
+  private object? ConvertDefaultValue(object value, AAEC.PropertyData.DataType dataType)
+  {
+    try
+    {
+      return dataType switch
+      {
+        AAEC.PropertyData.DataType.Integer => Convert.ToInt32(value),
+        AAEC.PropertyData.DataType.Real => Convert.ToDouble(value),
+        AAEC.PropertyData.DataType.TrueFalse => Convert.ToBoolean(value),
+        AAEC.PropertyData.DataType.Text => value.ToString(),
+        _ => value
+      };
+    }
+    catch (Exception ex) when (!ex.IsFatal())
+    {
+      _logger.LogWarning(ex, "Failed to convert default value {Value} to type {DataType}", value, dataType);
```

Comment:
Missing a `$` for the variable

Reply:
My two cents on this -> [here](https://github.com/specklesystems/speckle-sharp-connectors/pull/1149#discussion_r2435446247)